File: 1000-2000 LeetCode/1768.go
package main

import (
	"fmt"
)

func mergeStrings(word1 string, word2 string) string {
	r1 := []rune(word1)
	r2 := []rune(word2)

	len1 := len(r1)
	len2 := len(r2)

	merged := make([]rune, 0, len1+len2)

	i := 0
	for ; i < len1 && i < len2; i++ {
		merged = append(merged, r1[i])
		merged = append(merged, r2[i])
	}

	if i < len1 {
		merged = append(merged, r1[i:]...)
	} else if i < len2 {
		merged = append(merged, r2[i:]...)
	}

	return string(merged)
}

func main() {
	fmt.Println(mergeStrings("abc", "def"))
	fmt.Println(mergeStrings("abc", "defgh"))
	fmt.Println(mergeStrings("abcde", "def"))
}

File: 1000-2000 LeetCode/1768.py
def merge_strings(word1, word2) -> str:
  len1 = len(word1)
  len2 = len(word2)
  merged = []
  i=0
  while i < len1 and i < len2:
    merged.append(word1[i])
    merged.append(word2[i])
    i += 1
  
  if i < len1:
    merged.extend(word1[i:])
  elif i < len2:
    merged.extend(word2[i:])

  return ''.join(merged)

def main():
  print(merge_strings("abc", "def"))
  print(merge_strings("abc", "defgh"))

if __name__ == "__main__":
  main()